tests/Core/Config/SniffListTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Tests for the \PHP_CodeSniffer\Config --sniffs and --exclude arguments.
+ *
+ * @author  Dan Wallis <dan@wallis.nz>
+ * @license https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Config;
+
+use PHPUnit\Framework\TestCase;
+use PHP_CodeSniffer\Tests\ConfigDouble;
+
+/**
+ * Tests for the \PHP_CodeSniffer\Config --sniffs and --exclude arguments.
+ *
+ * @covers \PHP_CodeSniffer\Config::processLongArgument
+ */
+final class SniffListTest extends TestCase
+{
+
+
+    /**
+     * Ensure that the expected error message is returned for invalid arguments
+     *
+     * @param string $argument sniffs or exclude
+     * @param string $value    list of sniffs to include / exclude
+     * @param string $message  expected error message text
+     *
+     * @return       void
+     * @dataProvider dataInvalidSniffs
+     */
+    public function testInvalid($argument, $value, $message)
+    {
+        $config = new ConfigDouble();
+
+        $this->expectException('PHP_CodeSniffer\Exceptions\DeepExitException');
+        $this->expectExceptionMessage($message);
```

Comment:
While PHPCS still supports PHP < 5.6, this will need a toggle along the lines of the below:
```php
        if (method_exists($this, 'expectException') === true) {
            $this->expectException($exceptionClass);
            $this->expectExceptionMessage($exceptionMsg);
        } else {
            // PHPUnit < 5.2.0.
            $this->setExpectedException($exceptionClass, $exceptionMsg);
        }
```